### PR TITLE
Handle the case of TableNotFoundException in HBaseTableUtil#getTableStats

### PR DESCRIPTION
**Pull request by anwar6953**
Handle the case of TableNotFoundException when attempting to get the table descriptor of a RegionLoad (stats object), that was likely just recently deleted.

Example test case failure (stack trace) is below. Note that the referenced table (cdap_perm:priv) is actually created and deleted in the test case above 'testTableSizeMetrics'. So, likely, the NotFoundException is due to the cluster status still having reference to a table name that only recently has been deleted.
This test case wasn't flaky before because we didn't call `HBaseAdmin#getTableDescriptor` in this code path before (until https://github.com/caskdata/cdap/pull/8912).
```
HBase10CDHTableUtilTest testTableSizeMetrics History
Unit Tests    < 1 sec    
org.apache.hadoop.hbase.TableNotFoundException: cdap_perm:priv
    at org.apache.hadoop.hbase.client.HBaseAdmin.getTableDescriptor(HBaseAdmin.java:468)
    at co.cask.cdap.data2.util.hbase.HBaseTableUtil.getTableStats(HBaseTableUtil.java:609)
    at co.cask.cdap.data2.util.hbase.AbstractHBaseTableUtilTest.getTableStats(AbstractHBaseTableUtilTest.java:568)
    at co.cask.cdap.data2.util.hbase.AbstractHBaseTableUtilTest.testTableSizeMetrics(AbstractHBaseTableUtilTest.java:148)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
(42 more lines...)
```

### CODE REVIEW

**Comment by anwar6953** [comment]
Pending: create+delete table datasets (many times) on a cluster, and see how often this `TableNotFoundException` happens there…

File: cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java [hunk 1]
```diff
@@ -606,7 +607,14 @@ public DeleteBuilder buildDelete(Delete delete) {
 
       for (RegionLoad regionLoad : regionsLoad.values()) {
         TableName tableName = HRegionInfo.getTable(regionLoad.getName());
-        HTableDescriptor tableDescriptor = admin.getTableDescriptor(tableName);
+        HTableDescriptor tableDescriptor;
+        try {
+          tableDescriptor = admin.getTableDescriptor(tableName);
+        } catch (TableNotFoundException exception) {
+          // this can happen if the table has been deleted; the region stats get removed afterwards
+          LOG.warn("Table not found for table name {}. Skipping collecting stats for it.", tableName);
```
Comment (sreevatsanraman) [inline cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseTableUtil.java]: LOG the exception getMessage()

**Comment by sreevatsanraman** [comment]
Minor comment, fix and merge.

**Comment by anwar6953** [comment]
Build is green: https://builds.cask.co/browse/CDAP-RUT1053-1

**Pull request merged**